File: go_app/pkg/models/objects.go
package models

import (
	"encoding/json"
	"time"
)

type ObjConsumerStation struct {
	ID                 uint64              `gorm:"primaryKey" json:"id"`
	LocationDistrictId uint64              `json:"location_district_id"`
	LocationAreaId     uint64              `json:"location_area_id"`
	Name               string              `gorm:"unique" json:"name"`
	Address            string              `json:"address"`
	GeoData            GeoData             `gorm:"type:jsonb;default:'{}';" json:"geo_data"`
	Type               string              `json:"type"`
	PlaceType          string              `json:"place_type"`
	OdsName            string              `json:"ods_name"`
	OdsAddress         string              `json:"ods_address"`
	OdsIdUU            string              `json:"ods_id_uu"`
	OdsManagerCompany  string              `json:"ods_manager_company"`
	SourceStations     []*ObjSourceStation `gorm:"many2many:obj_source_consumer_stations" json:"source_stations"`
	Consumers          []ObjConsumer       `gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;" json:"consumers"`
}

type ObjSourceStation struct {
	ID                 uint64                `gorm:"primaryKey" json:"id"`
	LocationDistrictId uint64                `json:"location_district_id"`
	LocationAreaId     uint64                `json:"location_area_id"`
	Name               string                `gorm:"unique" json:"name"`
	Address            string                `json:"address"`
	EPower             int64                 `json:"e_power"`
	TPower             int64                 `json:"t_power"`
	BoilerCount        int64                 `json:"boiler_count"`
	TurbineCount       int64                 `json:"turbine_count"`
	LaunchedDate       time.Time             `json:"launched_date"`
	GeoData            GeoData               `gorm:"type:jsonb;default:'{}';" json:"geo_data"`
	ConsumerStations   []*ObjConsumerStation `gorm:"many2many:obj_source_consumer_stations" json:"consumer_stations"`
}

type ObjConsumer struct {
	ID                   uint64 `gorm:"primaryKey" json:"id"`
	ObjConsumerStationId uint64 `json:"obj_consumer_station_id"`
	LocationDistrictId   uint64 `json:"location_district_id"`
	LocationAreaId       uint64 `json:"location_area_id"`

	Address      string `gorm:"unique" json:"address"`
	Street       string `json:"street"`
	HouseType    string `json:"house_type"`
	HouseNumber  string `json:"house_number"`
	CorpusNumber string `json:"corpus_number"`
	SoorType     string `json:"soor_type"`
	SoorNumber   string `json:"soor_number"`

	BalanceHolder string `json:"balance_holder"`
	LoadGvs       string `json:"load_gvs"`
	LoadFact      string `json:"load_fact"`
	HeatLoad      string `json:"heat_load"`
	VentLoad      string `json:"vent_load"`

	GeoData   GeoData `gorm:"type:jsonb;default:'{}';" json:"geo_data"`
	TotalArea float64 `json:"total_area"`
	Target    string  `json:"target"`
	BClass    string  `json:"b_class"`
	Floors    int64   `json:"floors"`
	Number    string  `json:"number"`
	WearPct   string  `json:"wear_pct"`
	BuildYear int64   `json:"build_year"`

	Type          string `json:"type"`
	SockType      string `json:"sock_type"`
	EnergyClass   string `json:"energy_class"`
	OperatingMode string `json:"operating_mode"`
	Priority      int64  `json:"priority"`

	IsDispatch bool `json:"is_dispatch"`

	TempConditions TempCondition         `gorm:"type:jsonb;default:'{}';" json:"temp_conditions"`
	WeatherFall    []WeatherConsumerFall `json:"weather_fall"`
	EventsCounter  []EventCounter        `json:"events_counter"`
	Events         []EventConsumer       `json:"events"`
	WallMaterial   []*MaterialWall       `gorm:"many2many:material_consumer_walls" json:"wall_material"`
}

type TempCondition struct {
	SummerHigh float64 `json:"summer_high"`
	SummerLow  float64 `json:"summer_low"`
	WinterHigh float64 `json:"winter_high"`
	WinterLow  float64 `json:"winter_low"`
}

func (s *TempCondition) Scan(v interface{}) error {
	err := json.Unmarshal(v.([]byte), &s)
	if err != nil {
		return err
	}
	return nil
}

type GeoData struct {
	Polygon [][]float64 `json:"polygon"`
	Center  []float64   `json:"center"`
}

func (s *GeoData) Scan(v interface{}) error {
	err := json.Unmarshal(v.([]byte), &s)
	if err != nil {
		return err
	}
	return nil
}

File: pt_app/models/objects.py
from sqlalchemy import *
from sqlalchemy.orm import *
from models.utils import BaseModel, Base
from sqlalchemy.dialects.postgresql import JSONB

obj_source_consumer_stations = Table(
    'obj_source_consumer_stations', Base.metadata,
    Column('obj_source_station_id', Integer(), ForeignKey('obj_source_stations.id'), primary_key=True),
    Column('obj_consumer_station_id', Integer(), ForeignKey('obj_consumer_stations.id'), primary_key=True)
)


class ObjConsumerStation(BaseModel):
    name: str
    address: str

    __tablename__ = "obj_consumer_stations"

    id = Column(Integer, primary_key=True, autoincrement=True)
    location_district_id = Column(Integer, ForeignKey('location_districts.id'), nullable=True)
    location_area_id = Column(Integer, ForeignKey('location_areas.id'), nullable=True)

    name = Column(String, nullable=True)
    address = Column(String, nullable=True)
    geo_data = Column(JSONB, nullable=True, default='{}')
    type = Column(String, nullable=True)
    place_type = Column(String, nullable=True)
    ods_name = Column(String, nullable=True)
    ods_address = Column(String, nullable=True)
    ods_id_uu = Column(String, nullable=True)
    ods_manager_company = Column(String, nullable=True)

    source_stations = relationship('ObjSourceStation', uselist=True, secondary='obj_source_consumer_stations',
                                   lazy='dynamic')
    consumers = relationship('ObjConsumer', uselist=True, lazy='dynamic')

    UniqueConstraint(name, name='uni_obj_consumer_stations_name')
    # UniqueConstraint(address, name='uni_obj_consumer_stations_address')


class ObjSourceStation(BaseModel):
    name: str
    address: str

    __tablename__ = "obj_source_stations"

    id = Column(Integer, primary_key=True, autoincrement=True)

    location_district_id = Column(Integer, ForeignKey('location_districts.id'), nullable=True)
    location_area_id = Column(Integer, ForeignKey('location_areas.id'), nullable=True)

    name = Column(String, nullable=True)
    address = Column(String, nullable=True)
    geo_data = Column(JSONB, nullable=True, default='{}')

    e_power = Column(String, nullable=True)
    t_power = Column(BigInteger, nullable=True)
    boiler_count = Column(BigInteger, nullable=True)
    turbine_count = Column(BigInteger, nullable=True)
    launched_date = Column(DateTime(timezone=True), nullable=True)

    consumer_stations = relationship('ObjConsumerStation', uselist=True, secondary='obj_source_consumer_stations',
                                     lazy='dynamic')

    UniqueConstraint(name, name='uni_obj_source_stations_name')


class ObjConsumer(BaseModel):
    __tablename__ = "obj_consumers"

    id = Column(Integer, primary_key=True, autoincrement=True)
    obj_consumer_station_id = Column(Integer, ForeignKey('obj_consumer_stations.id'), nullable=True)
    location_district_id = Column(Integer, ForeignKey('location_districts.id'), nullable=True)
    location_area_id = Column(Integer, ForeignKey('location_areas.id'), nullable=True)

    address = Column(String, nullable=True)
    geo_data = Column(JSONB, nullable=True, default='{}')

    street = Column(String, nullable=True)
    house_type = Column(String, nullable=True)
    house_number = Column(String, nullable=True)
    corpus_number = Column(String, nullable=True)
    soor_type = Column(String, nullable=True)
    soor_number = Column(String, nullable=True)

    balance_holder = Column(String, nullable=True)
    load_gvs = Column(String, nullable=True)
    load_fact = Column(String, nullable=True)
    heat_load = Column(String, nullable=True)
    vent_load = Column(String, nullable=True)

    total_area = Column(Float, nullable=True)
    target = Column(String, nullable=True)
    b_class = Column(String, nullable=True)
    floors = Column(String, nullable=True)
    number = Column(String, nullable=True)
    wear_pct = Column(String, nullable=True)
    build_year = Column(String, nullable=True)
    type = Column(String, nullable=True)
    sock_type = Column(String, nullable=True)
    energy_class = Column(String, nullable=True)
    operating_mode = Column(String, nullable=True)
    priority = Column(Integer, nullable=True)
    is_dispatch = Column(Boolean, nullable=True)

    temp_conditions = Column(JSONB, nullable=True, default='{}')
    events = relationship('EventConsumer', uselist=True, lazy='dynamic')
    weather_fall = relationship('WeatherConsumerFall', uselist=True, lazy='dynamic')
    wall_material = relationship('MaterialWall', uselist=True, secondary='material_consumer_walls',
                                 lazy='dynamic')

    UniqueConstraint(address, name='uni_obj_consumers_address')
